### PR TITLE
fix(gen2): Add auth overrides section

### PR DESCRIPTION
**Pull request by ErikCH**
Added section on cognito auth overrides

### CODE REVIEW

File: src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx [hunk 1]
```diff
@@ -21,12 +21,65 @@ import { defineAuth } from '@aws-amplify/backend';
...
+  loginWith: {
+    email: {
+      verificationEmailStyle: "CODE",
+      verificationEmailBody: (code: string) =>
+        `Welcome! Your verification code is ${code}.`,
+      verificationEmailSubject: "Welcome! Here is your verification code",
+    },
+  },
+});
+```
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx:47]: I like this example as a form of "customizing sign-up flow", can we lift this to https://github.com/aws-amplify/docs/blob/next-release/gen2/src/pages/gen2/build-a-backend/auth/enable-sign-up/index.mdx ?

File: src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx [hunk 2]
```diff
@@ -40,9 +93,11 @@ const backend = defineBackend({
 // extract L1 CfnUserPool resources
 const { cfnUserPool } = backend.resources.auth.resources.cfnResources;
 // use CDK's `addPropertyOverride` to modify properties directly
-cfnUserPool.addPropertyOverride('Policies.PasswordPolicy.MinimumLength', 32);
+cfnUserPool.addPropertyOverride("Schema", [
+  {
+    Name: "publicName",
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx:98]: can we show a custom or dev attribute example? that's one thing that's not available in `defineAuth`'s `userAttributes` prop

File: src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx [hunk 3]
```diff
@@ -21,12 +21,65 @@ import { defineAuth } from '@aws-amplify/backend';
...
+      MinimumLength: 10,
+      RequireLowercase: true,
+      RequireNumbers: true,
+      RequireSymbols: true,
+      RequireUppercase: true,
+      TemporaryPasswordValidityDays: 20,
+    },
+  }
+);
+```
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx:77]:
```suggestion
```ts title="amplify/backend.ts"
// amplify/backend.ts
import { defineBackend } from '@aws-amplify/backend';
import { auth } from './auth/resource.js';

const backend = defineBackend({
  auth,
});

// extract L1 CfnUserPool resources
const { cfnUserPool } = backend.resources.auth.resources.cfnResources;
// use CDK's `addPropertyOverride` to modify properties directly
cfnUserPool.addPropertyOverride(
  "Policies",
  {
    PasswordPolicy: {
      MinimumLength: 10,
      RequireLowercase: true,
      RequireNumbers: true,
      RequireSymbols: true,
      RequireUppercase: true,
      TemporaryPasswordValidityDays: 20,
    },
  }
);
```
```

Comment (josefaidt) [reply src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx:77]: few adjustments on this one:
- Backend -> defineBackend
- copy/pasted comments from the other example
- add a comment with the filename at the top of the example

File: src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx [hunk 4]
```diff
@@ -21,12 +21,65 @@ import { defineAuth } from '@aws-amplify/backend';
...
+
+## Custom Attributes
+
+The following code will allow you to add custom attributes using the [schema](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-cognito.CfnUserPool.html#schema) property with the L1 `cfnUserPool` construct.
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx:81]:
```suggestion
The following code will allow you to add custom attributes using the [UserPool schema](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-cognito.CfnUserPool.html#schema) property with the L1 `cfnUserPool` construct.
```

File: src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx (hunk 2 above)
Comment (josefaidt) [reply src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx:98]: does this overwrite the existing schema (with `email`)?

File: src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx (hunk 1 above)
Comment (ErikCH) [reply src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx:47]: OK I moved it to the `enable-sign-up` section

File: src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx (hunk 3 above)
Comment (ErikCH) [reply src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx:77]: Will do

File: src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx (hunk 4 above)
Comment (ErikCH) [reply src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx:81]: 👍

File: src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx (hunk 2 above)
Comment (ErikCH) [reply src/pages/gen2/build-a-backend/auth/override-cognito/index.mdx:98]: Overriding `Schema` will add custom attributes. When I updated the `Schema` in this example, it didn't override the existing email updates I did. If there is an easier way of updating custom attributes let me know.

**Comment by ErikCH** [comment]
Closing this in favor of #6516

**Pull request closed without merging**